sdv/metadata/table.py

```diff
@@ -513,10 +525,17 @@ def _fit_hyper_transformer(self, data, extra_columns):
 
         transformers_dict = self._get_transformers(dtypes)
         for column in numerical_extras:
-            transformers_dict[column] = rdt.transformers.NumericalTransformer()
+            transformers_dict[column] = rdt.transformers.FloatFormatter()
+
+        self._hyper_transformer = rdt.HyperTransformer()
+        self._hyper_transformer.detect_initial_config(data)
+        if transformers_dict:
+            self._hyper_transformer.update_transformers(transformers_dict)
 
-        self._hyper_transformer = rdt.HyperTransformer(field_transformers=transformers_dict)
-        self._hyper_transformer.fit(data[list(transformers_dict.keys())])
+        try:
+            self._hyper_transformer.fit(data)
+        except rdt.errors.Error:
```

Comment:
why do we need to catch errors here?

Reply:
This error was produced because the `data` was empty, we check now if the `data` is empty we just don't fit.